desc/magnetic_fields.py

```diff
@@ -1094,3 +1129,614 @@ def odefun(rpz, s):
+            (ie, e_theta x e_zeta points outward from the surface).
+
+        """
+        if not isinstance(surface, FourierRZToroidalSurface):
+            raise TypeError(
+                "Expected type FourierRZToroidalSurface for argument surface, "
+                f"instead got type {type(surface)}"
+            )
+        R_lmn = surface.R_lmn
+        Z_lmn = surface.Z_lmn
+        modes_R = surface._R_basis.modes[:, 1:]
+        modes_Z = surface._Z_basis.modes[:, 1:]
+        NFP = surface.NFP
+        sym = surface.sym
+        rho = surface.rho
+        name = surface.name
+
+        return cls(
+            Phi_mn,
+            modes_Phi,
+            I,
+            G,
+            sym_Phi,
+            R_lmn,
+            Z_lmn,
+            modes_R,
+            modes_Z,
+            NFP,
+            sym,
+            rho,
+            name,
+            check_orientation=False,
+        )
+
+
+def _compute_magnetic_field_from_CurrentPotentialField(
+    field, coords, params=None, basis="rpz", grid=None
+):
+    """Compute magnetic field at a set of points.
+
+    Parameters
+    ----------
+    field : CurrentPotentialField or FourierCurrentPotentialField
+        current potential field object from which to compute magnetic field.
+    coords : array-like shape(N,3) or Grid
+        cylindrical or cartesian coordinates
+    params : dict, optional
+        parameters to pass to compute function
+        should include the potential
+    basis : {"rpz", "xyz"}
+        basis for input coordinates and returned magnetic field
+    grid : Grid,
+        source grid upon which to evaluate the surface current density K
+
+    Returns
+    -------
+    field : ndarray, shape(N,3)
+        magnetic field at specified points
+
+    """
+    assert basis.lower() in ["rpz", "xyz"]
+    if isinstance(coords, Grid):
+        coords = coords.nodes
+    coords = jnp.atleast_2d(coords)
+    if basis == "rpz":
+        coords = rpz2xyz(coords)
+    surface_grid = grid or LinearGrid(M=30, N=30, NFP=field.NFP)
+
+    # compute surface current, and store grid quantities
+    # needed for integration in class
+    # TODO: does this have to be xyz, or can it be computed in rpz as well?
+    data = field.compute(["K", "x"], grid=surface_grid, basis="xyz", params=params)
+    _K = xyz2rpz_vec(data["K"], phi=surface_grid.nodes[:, 2])
+    _rs = xyz2rpz_vec(data["x"], phi=surface_grid.nodes[:, 2])
+    # surface element, must divide by NFP to remove the NFP multiple on the
+    # surface grid weights, as we account for that when doing the for loop
+    # over NFP
+    _dV = surface_grid.weights * data["|e_theta x e_zeta|"] / surface_grid.NFP
+
+    def nfp_loop(j, f):
+        # calculate (by rotating) rs, rs_t, rz_t
+        phi = (surface_grid.nodes[:, 2] + j * 2 * jnp.pi / surface_grid.NFP) % (
+            2 * jnp.pi
+        )
+        rs = rpz2xyz_vec(_rs, phi=phi)
```

Comment:
same as above. To rotate this you just need to add `j*2*np.pi/NFP` to the phi coordinate